src/pspm_overwrite.m

```diff
@@ -0,0 +1,89 @@
+function ow_final = pspm_overwrite(varargin)
+
+% DESCRIPTION
+% pspm_overwrite generalises the overwriting operation
+% pspm_overwrite considers the following situations
+% - whether options.overwrite is defined
+% - whether PsPM is in develop mode
+% - whether the file exist
+% - whether GUI can be used
+% Only the following situation will stop PsPM to overwrite files
+% 1. if overwrite is defined as not to overwrite
+% 2. if overwrite is not defined, and users use the GUI to stop overwriting
+
+% ARGUMENTS
+% fn        the name of the file to possibly overwrite
+%           can be a link if necessary
+% ow        option of overwrite if this option is presented
+%           can be a value or a struct
+%           if a value, can be 0 (not to overwrite) or 1 (to overwrite)
+%           if a struct, check if the field "overwrite" exist
+
+% OUTPUTS
+% ow_final  option of overwriting determined by pspm_overwrite
+
+pth = fileparts(which('pspm'));
+settings_struct = load([pth, '/pspm_settings.mat'], "settings");
+settings = settings_struct.settings;
+
+
+
+%% start to define ow
+switch numel(varargin)
+  case 0
+    warning('ID:invalid_input', 'at least one argument is required');
+    return;
+  case 1 % ow is not defined
+    fn = varargin{1};
+    if iscell(fn)
+      fn = fn{1};
+    end
+    switch settings.developmode
+      case 1
+        ow_final = 1; % In develop mode, always overwrite
+      otherwise
```

Comment:
It is a bit confusing to use a switch for a single case.
Are you planning to add multiple `settings.developmode` in the future?
If so, please discard this comment.

Reply:
> It is a bit confusing to use a switch for a single case.
> Are you planning to add multiple `settings.developmode` in the future?
> If so, please discard this comment.

I wrote it like that because `settings.developmode` may be defined as something else and I want to avoid such bugs... However I agree it is a little confusing. I have changed it to logical values now.